test/integration/user-management-api/applications_test.rb

```diff
@@ -17,10 +17,11 @@ def setup
 
     @application_plan = FactoryBot.create(:application_plan, issuer: @provider.default_service)
     @application_plan.publish!
-    @buyer.buy! @application_plan
-
     @application = @buyer.buy! @application_plan
 
+    # Create additional apps to have more data
+    @buyer.buy! @application_plan
```

Comment:
Did you figure out what is the difference between 
```
@application = @buyer.buy! @application_plan
@buyer.buy! @application_plan
```
and
```
@buyer.buy! @application_plan
@application = @buyer.buy! @application_plan
```
and why it works one way but not the other?

Reply:
It's the same, I just changed it for a better readability. One of the apps `@application` is used in the tests, so I moved its declaration to the top. The second one is just to have some more data, so I moved it to the bottom. But they do the same.